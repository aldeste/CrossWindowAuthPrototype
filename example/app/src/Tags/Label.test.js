import Label from "./Label";
import renderer from "react-test-renderer";
import React from "react";
import "jest-styled-components";

describe("Label tag", () => {
  it("Should be defined", () => {
    expect(Label).toBeDefined();
  });

  it("Is a styled component", () => {
    expect(Label.name).toBe("StyledComponent");
  });

  it("Matches earlier configuration", () => {
    const tree = renderer.create(<Label />).toJSON();
    expect(tree).toMatchStyledComponentsSnapshot();
  });

  it("Matches earlier configuration if required is passed", () => {
    const tree = renderer.create(<Label required />).toJSON();
    expect(tree).toMatchStyledComponentsSnapshot();
  });
});
