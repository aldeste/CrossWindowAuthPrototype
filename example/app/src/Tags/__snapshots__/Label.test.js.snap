// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label tag Matches earlier configuration 1`] = `
.iCwAoZ {
  font-family: inherit;
  font-weight: 600;
}

.iCwAoZ + input {
  margin-top: 0;
}

<label
  className="Label__Label-jsURsk iCwAoZ"
/>
`;

exports[`Label tag Matches earlier configuration if required is passed 1`] = `
.fOlxVT {
  font-family: inherit;
  font-weight: 600;
}

.fOlxVT + input {
  margin-top: 0;
}

.fOlxVT::after {
  content: '*';
  margin: 0 0.1em;
  color: hsl(343,88%,53%);
}

<label
  className="Label__Label-jsURsk fOlxVT"
  required={true}
/>
`;
